systemd/CMakeLists.txt

```
@@ -1,2 +1,2 @@
 install(FILES pimodbus-master.service pimodbus-slave.service
-	DESTINATION /lib/systemd/system)
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/systemd/system)
```

Did you test this? In the docs they say something like
```
LIBDIR
object code libraries (lib or lib64)

On Debian, this may be lib/<multiarch-tuple> when [CMAKE_INSTALL_PREFIX](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html#variable:CMAKE_INSTALL_PREFIX) is /usr.
```
So now it ist something like `/usr/lib/<multiarch-tuple>/systemd/system`, but what is <multiarch-tuple>?

https://wiki.debian.org/Multiarch/Tuples

---

On an aarch64 ${CMAKE_INSTALL_LIBDIR} is `/usr/lib/aarch64-linux-gnu`. I think this is the wrong place for systemd. 🙁

---

```suggestion
	DESTINATION ${CMAKE_INSTALL_LIBDIR}/../systemd/system)
```
Without a joke, that's how it would work. I just built the Debian package and it will now be installed in `/usr/lib/systemd/system`.

---

```suggestion
	DESTINATION ${CMAKE_INSTALL_LIBDIR}/../systemd/system)
```
Without a joke, that's how it would work. I just built the Debian package and it will now be installed in `/usr/lib/systemd/system`.

---

I tested this locally:

```
$ cmake --install . --prefix pkg
-- Install configuration: ""
-- Installing: /home/tboehler/dev/revpi-modbus/build/pkg/sbin/piModbusMaster
-- Installing: /home/tboehler/dev/revpi-modbus/build/pkg/sbin/piModbusSlave
-- Installing: /home/tboehler/dev/revpi-modbus/build/pkg/lib/systemd/system/pimodbus-master.service
-- Installing: /home/tboehler/dev/revpi-modbus/build/pkg/lib/systemd/system/pimodbus-slave.service
$ tree pkg
pkg
├── lib
│   └── systemd
│       └── system
│           ├── pimodbus-master.service
│           └── pimodbus-slave.service
└── sbin
    ├── piModbusMaster
    └── piModbusSlave

5 directories, 4 files
```

> On an aarch64 ${CMAKE_INSTALL_LIBDIR} is /usr/lib/aarch64-linux-gnu. I think this is the wrong place for systemd.

Good catch, I didn't think that would be the case. I'll revert the change for the `.service` files but keep it for the others.

---

TIL that the `--prefix` flag *completely* overrides the `/usr/local` prefix normally set by CMake. This wouldn't have worked for the `.service` files anyway, sorry!
I assumed it would be used like `DESTDIR`: https://www.gnu.org/prep/standards/html_node/DESTDIR.html

---

For systemd we should use something like this: https://github.com/ximion/limba/blob/master/data/cmake/systemdservice.cmake
I'm not sure on how to integrate it the best way.

---

Integrating it into just this repository doesn't make much sense as other CMake projects would benefit from this, too, so ideally this should be available everywhere. I'm not sure if this is worth the hassle to integrate in some way, though.

---

As we use udev rules in different packages it would be good to have a correct and unified way to do this. But as you said it might not be "worth the hassle". Let me think about it, this is **no** blocker for this PR!